_populate_db/populate.py

```diff
@@ -1,20 +1,59 @@
 import os, sys, re
 import numpy as np
 from xml.etree import ElementTree
-import dicom
+try:
+    import dicom
+except ImportError:
+    import pydicom
 
 assert not os.path.exists(os.path.join(os.path.pardir, 'pylidc.sqlite')), "`pylidc.sqlite` already exists. Aborting."
```

Comment:
This line doesn't work as intended. The database will always exist since it is out on pypi now. Maybe we should replace this with some simple *Are you sure you want to continue?* type of check at the beginning.

Reply:
Note that line of code was not changed in my PR - it was there originally. So I do not know for sure how this line was intended to work.

My interpretation was that `populate.py` is run for the local installation as a way to generate the database. The database exists in the github repo, so I think the code makes that assumption that the `populate` script is executed from the source repo without the pypi installed content.

From my perspective, it would be nice to allow configuring the code to either use the default database that comes from pypi, or newly generated database that advanced users might want to populate from the original dataset.

Reply:
Sorry I didn't mean to come off as accusing you of breaking that line. This was just as much a note to myself :grinning: 

At this point, the database always comes with the source -- whether through github or pypi. Maybe you can explain more what the use-case is. It would not be possible to allow for switching to a new database with new features because the database is inherently linked to the sqlalchemy object models (i.e., `Scan`, `Annotation`, etc.). This would require adding new attributes to those classes, which of course couldn't be done some database switching logic alone. If this is your intention, I suggest forking the repo, or if you'd like to add it in this repo, raising an issue for a feature request that you are looking for specifically, so that we consider adding that.

Reply:
No problem! 👍 

Thank you for your explanation. In my case, what I wanted to do is to add the `SOPInstanceUID` for individual annotation contours here: https://github.com/pylidc/pylidc/blob/master/_populate_db/populate.py#L158. I am actually not sure why that was not included originally.

Somewhat unrelated to this (and sorry for veering off topic of this PR, please let me know if I should post this as a separate issue/question), can you explain why you implemented the [`cluster_annotations`](https://github.com/pylidc/pylidc/blob/master/pylidc/Scan.py#L306) heuristic? Isn't it more appropriate to use `noduleID` for associating individual contours with a single nodule?

Reply:
The attributes that were included were was that I chose to include. So, that's why SOPInstanceUID was not included: I never used it. Note that the scan objects have `series_instance_uid` and `study_instance_uid` parameters. The SOPInstanceUID would be easy to add if the populate script was more dependable.

Regarding the `cluster_annotations` method: `noduleID` doesn't do what you think it does (as far as I can tell) for grouping annotations that refer to the same physical nodule in a scan. Example:

```
import pylidc as pl

scan = pl.query(pl.Scan).first()

for i, nodule in enumerate(nodules):
    print("Nodule {}".format(i+1))
    print("-"*80)
    for annotation in nodule:
        print(annotation._nodule_id)
    print("")
```
prints:
```
Nodule 1
--------------------------------------------------------------------------------
4
12329
16317
2054

Nodule 2
--------------------------------------------------------------------------------
3
12325
16313
2050

Nodule 3
--------------------------------------------------------------------------------
16332

Nodule 4
--------------------------------------------------------------------------------
6
12321
16309
2046
```
I have no idea what the intended purpose of `noduleID`, but it (frustratingly) does not group annotations as the name suggests.

Reply:
For the sake of completeness, I assume the correct version of your code is below (one line was missing):
```python
import pylidc as pl

scan = pl.query(pl.Scan).first()
nodules = scan.cluster_annotations()

for i, nodule in enumerate(nodules):
    print("Nodule {}".format(i+1))
    print("-"*80)
    for annotation in nodule:
        print(annotation._nodule_id)
    print("")
```

Related to this, do you know why if I get the following error if I try to access `annotation.boolean_mask().shape` for nodule 6? Is this a bug or not? (this is subject LIDC-IDRI-0078)

![image](https://user-images.githubusercontent.com/313942/42399300-4b9a8728-813b-11e8-99d5-f0377d74a29f.png)

Reply:
are you on the pypi version or the github version of pylidc?

Reply:
I believe I am on the pypi version.

Reply:
The 0.2 release was a major refactor. There were a few bugs that I fixed, but didn't publish to pypi yet. You can try the github version in the meantime.

Reply:
Indeed, this problem does not happen when running from the master-based install. Thanks!